source/jormungandr/jormungandr/interfaces/v1/serializer/base.py

```diff
@@ -80,3 +79,62 @@ class GenericSerializer(PbNestedSerializer):
     id = serpy.Field(attr='uri')
     name = serpy.Field()
 
+
+class LiteralField(serpy.Field):
+    """
+    :return literal value
+    """
+    def __init__(self, value, *args, **kwargs):
+        super(LiteralField, self).__init__(*args, **kwargs)
+        self.value = value
+
+    def as_getter(self, serializer_field_name, serializer_cls):
+        return lambda *args, **kwargs: self.value
+
+
+def flatten(obj):
+    if type(obj) != dict:
+        raise ValueError("Invalid argument")
+    new_obj = {}
+    for key, value in obj.items():
+        if type(value) == dict:
+            tmp = {'.'.join([key, _key]): _value for _key, _value in flatten(value).items()}
+            new_obj.update(tmp)
+        else:
+            new_obj[key] = value
+    return new_obj
+
+
+def value_by_path(obj, path, default=None):
+    new_obj = flatten(obj)
+    if new_obj and path in new_obj:
+        return new_obj[path]
+    else:
+        return default
+
+
+def str_to_int(value, default=None):
+    try:
+        return int(value)
+    except:
+        return default
+
+
+class NestedPropertyField(serpy.Field):
+    def as_getter(self, serializer_field_name, serializer_cls):
+        return lambda v: self.get_property(v, self.attr)
+
+    def get_property(self, obj, path):
+        return value_by_path(obj, path)
+
+
+class IntNestedPropertyField(serpy.Field):
```

Comment:
can't you just make it inherit from `NestedPropertyField` and say `to_value = staticmethod(int)` ?